pkg/printers/internalversion/printers.go

```diff
@@ -807,17 +808,21 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 					lastRestartDate = terminatedDate
 				}
 			}
-			if container.State.Waiting != nil && container.State.Waiting.Reason != "" {
+			switch {
+			case container.State.Waiting != nil && container.State.Waiting.Reason != "":
 				reason = container.State.Waiting.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
-				reason = container.State.Terminated.Reason
-			} else if container.State.Terminated != nil && container.State.Terminated.Reason == "" {
-				if container.State.Terminated.Signal != 0 {
-					reason = fmt.Sprintf("Signal:%d", container.State.Terminated.Signal)
+			case container.State.Terminated != nil:
+				if container.State.Terminated.ExitCode == 0 && len(container.State.Terminated.Reason) != 0 {
+					reason = container.State.Terminated.Reason
 				} else {
-					reason = fmt.Sprintf("ExitCode:%d", container.State.Terminated.ExitCode)
+					hasError = true
+					if container.State.Terminated.Signal != 0 {
```

Comment:
if the container.State.Terminated.Reason is not empty, shall we ignore it? Also if hasError, the reason would be overrided below, is this block neccessary?

Reply:
You are right. 
Then i add  a function `hasErrorReason`. If reason has error message. It would not overwrite.